src/test/java/seedu/address/logic/commands/commandtestutils/FlashcardBankCommandTestUtil.java

```diff
@@ -57,4 +63,23 @@ public static void assertCommandSuccess(Command<FlashcardModel> command, Flashca
         CommandResult expectedCommandResult = new CommandResult(expectedMessage);
         assertCommandSuccess(command, actualModel, expectedCommandResult, expectedModel);
     }
+
+    /**
+     * Executes the given {@code command}, confirms that <br>
+     * - a {@code CommandException} is thrown <br>
+     * - the CommandException message matches {@code expectedMessage} <br>
+     * - the flashcard bank, filtered flashcard set list and selected flashcard set <br>
+     * - in {@code actualModel} remain unchanged
+     */
+    public static void assertCommandFailure(Command<FlashcardModel> command, FlashcardModel actualModel,
+                                            String expectedMessage) {
+        // we are unable to defensively copy the model for comparison later, so we can
+        // only do so by copying its components.
+        FlashcardBank expectedFlashcardBank = new FlashcardBank(actualModel.getFlashcardBank());
+        List<FlashcardSet> expectedFilteredList = new ArrayList<>(actualModel.getFlashcardSetList());
+
+        assertThrows(CommandException.class, expectedMessage, () -> command.execute(actualModel));
+        assertEquals(expectedFlashcardBank, actualModel.getFlashcardBank());
+        assertEquals(expectedFilteredList, actualModel.getFlashcardSetList());
```

Comment:
What is the purpose of the `assertEquals` here? They do not use the `command` in the parameter. When would the 2 `assertEquals` return False in this case?

Reply:
I think it's an extra check to see if the contents in flashcard bank changes even when the error is thrown. So it will return false when the flashcard bank/flashcard set list is modified